soh/src/code/z_player_lib.c

```diff
@@ -1615,24 +1991,398 @@ void func_80090D20(PlayState* play, s32 limbIndex, Gfx** dList, Vec3s* rot, void
+
+                            CLOSE_DISPS(play->state.gfxCtx);
+                        }
+                        if (LINK_IS_CHILD) {
+                            Matrix_Scale(1.25f, 1.25f, 1.25f, MTXMODE_APPLY);
+                        }
+                        break;
+                    case PLAYER_SWORD_BGS:
+                        OPEN_DISPS(play->state.gfxCtx);
+
+                        if (LINK_IS_CHILD) {
+                            Matrix_Scale(0.8f, 0.8f, 0.8f, MTXMODE_APPLY);
+                        }
+
+                        gSPMatrix(POLY_OPA_DISP++, MATRIX_NEWMTX(play->state.gfxCtx),
+                                  G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+                        gSPDisplayList(POLY_OPA_DISP++, (Gfx*)gLinkLongswordSheathDL);
+
+                        CLOSE_DISPS(play->state.gfxCtx);
+                        if (this->leftHandType != 4) {
+                            OPEN_DISPS(play->state.gfxCtx);
+
+                            gSPMatrix(POLY_OPA_DISP++, MATRIX_NEWMTX(play->state.gfxCtx),
+                                      G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+                            gSPDisplayList(
+                                POLY_OPA_DISP++, (Gfx*)gLinkLongswordInSheathDL);
+
+                            CLOSE_DISPS(play->state.gfxCtx);
+                        }
+                        if (LINK_IS_CHILD) {
+                            Matrix_Scale(1.25f, 1.25f, 1.25f, MTXMODE_APPLY);
+                        }
+                    }
+                if (this->rightHandType != 10) {
+                    switch (CUR_EQUIP_VALUE(EQUIP_SHIELD)) {
+                        case PLAYER_SHIELD_DEKU:
+                            OPEN_DISPS(play->state.gfxCtx);
+
+                            if (LINK_IS_ADULT) {
+                                Matrix_Scale(1.2f, 1.2f, 1.2f, MTXMODE_APPLY);
+                            }
+
+                            gSPMatrix(POLY_OPA_DISP++, MATRIX_NEWMTX(play->state.gfxCtx),
+                                      G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+                            gSPDisplayList(POLY_OPA_DISP++, (Gfx*)gLinkDekuShieldOnBackDL);
+
+                            CLOSE_DISPS(play->state.gfxCtx);
+                            break;
+                        case PLAYER_SHIELD_HYLIAN:
+                            if (!Player_IsChildWithHylianShield(this)) {
+                                OPEN_DISPS(play->state.gfxCtx);
+
+                                if (LINK_IS_CHILD) {
+                                    Matrix_Scale(.80f, .80f, .80f, MTXMODE_APPLY);
+                                }
+
+                                gSPMatrix(POLY_OPA_DISP++, MATRIX_NEWMTX(play->state.gfxCtx),
+                                          G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+                                gSPDisplayList(POLY_OPA_DISP++, (Gfx*)gLinkHylianShieldOnBackDL);
```

Comment:
I'm noticing this pattern of function calls seems to be in a lot of places. In order to make things a bit cleaner, could you make a function like `Player_DrawItemChildScaled(float childScale, Gfx* dlist)` that contains this block? So for the hylian shield here this block would be condensed down to just `Player_DrawItemChildScaled(0.80f, gLinkHylianShieldOnBackDL);`.

Reply:
Updated that. The scale isn't an input, but the scale is now the proper scale for adult to child and back, 1.35 instead of 1.25, though it's not as easily inversible.